packages/flutter_tools/lib/src/asset.dart

```diff
@@ -938,10 +969,48 @@ class ManifestAssetBundle implements AssetBundle {
       packageName,
       attributedPackage,
       assetKind: assetKind,
+      originUri: originUri,
+      flavor: flavor,
     );
+
+    final bool hasMoreSpecificDeclarationWithFlavor = ((){
+      final _Asset? preExistingAsset = result.keys
+        .where((_Asset other) => other.entryUri == asset.entryUri)
+        .firstOrNull;
+      if (preExistingAsset == null) {
+        return false;
+      }
+
+      if (flavor != null && flavor != preExistingAsset.flavor ) {
+        if (assetUri.path.endsWith('/') == preExistingAsset.originUri.path.endsWith('/')) {
+          throwToolExit('Asset "${assetUri.path}" belongs to multiple flavors.\n'
+            'An entry with the path "${preExistingAsset.originUri.path}" gives it a flavor of "${preExistingAsset.flavor}".\n'
+            'An entry with the path "${assetUri.path}" gives it a flavor of "$flavor".'
+          );
+        }
+        if (assetUri.path.endsWith('/') && !preExistingAsset.originUri.path.endsWith('/')) {
+          return true;
+        }
+        else if (!assetUri.path.endsWith('/') && preExistingAsset.originUri.path.endsWith('/')) {
+          _logger.printTrace('Skipping assets entry "${asset.entryUri.path}" since its configured flavor, "${asset.flavor}", did not match the provided flavor (if any).');
+          result.remove(preExistingAsset);
+        }
+      }
+
+      return false;
+    })();
+
+    if (hasMoreSpecificDeclarationWithFlavor) {
+      _logger.printTrace('Skipping assets entry "${asset.entryUri.path}" since its configured flavor, "${asset.flavor}", did not match the provided flavor (if any).');
+      return;
+    }
+
     final List<_Asset> variants = <_Asset>[];
     final File assetFile = asset.lookupAssetFile(_fileSystem);
-    for (final String path in cache.variantsFor(assetFile.path)) {
+
+    final List<String> relevantFiles =  cache.variantsFor(assetFile.path);
+
+    for (final String path in relevantFiles) {
```

Comment:
?

Reply:
Indeed. This was probably an artifact of some earlier experiment. Fixed in [7ae8ba2](https://github.com/flutter/flutter/pull/132985/commits/7ae8ba2f8e304ae81dcb8a72a927200d12d87de2)